src/index.ts

```diff
@@ -308,23 +317,29 @@ export class Paginator {
     });
 
     function makeRequest(query?: ParsedArguments|string) {
-      originalMethod(query, onResultSet);
```

Comment:
Is this change actually needed? Unless we have plans to drop callback support entirely, I think it is probably safe to leave this alone, no?

Reply:
I made this change in order for `paginator` not to be dependent on `callbackify` and be able to handle a truly pure promise base function.
Also in hopes, that it will be more efficient this way rather than funnel every function through `callbackify` every time.

Let me know your preference though.

Reply:
Won't `parsedArguments.callback` be the user provided callback function though? Using the user function to determine which version (promise/callback) to call means that you will definitely be dependent on callbackify for promise first clients. I think if your goal is to not be dependent on callbackify/promisify you should probably always use the promise version under the assumption that callback first will eventually go away.

Reply:
I guess what I wanted to say is, in case when user didn't provide callback in the original method, I didn't want `paginator` to provide one. This way `callbackify` will just execute original method as is, instead of going through the process of removing imposed `cb` from `arguments`, still executing as promise and pass the `arrified` response to callback.
And also in remote situation when someone will use `paginator` to extend just a promise (not callbackified) it wouldn't crush.

Reply:
Using the streaming APIs as an example, `parsedArguments.callback` will always be`undefined`, right? This logic then assumes that since there is no callback, the original method must be async first, yes?

Reply:
Got it, done
thanks

Reply:
Oh man, sorry if I've been a pain in the butt! I liked the optimization, I would have just preferred an option or something to indicate which version to use. Maybe we should open an issue that says to convert this to promise only after all the clients have been updated.